core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -37,6 +43,100 @@ class HttpSamDAO(baseSamServiceURL: String)(implicit val system: ActorSystem) ex
     }
   }
 
+  override def createResource(resourceTypeName: SamResourceTypeName, resourceId: String, userInfo: UserInfo): Future[Boolean] = {
+    val url = samServiceURL + s"/api/resource/${resourceTypeName.value}/$resourceId"
+    val httpRequest = Post(url)
+    val pipeline = addAuthHeader(userInfo) ~> sendReceive
+    val result: Future[HttpResponse] = pipeline(httpRequest)
+
+    retry(when500) { () =>
+      result.map { response =>
+        response.status match {
+          case s if s.isSuccess => true
+          case _ => false
+        }
+      }
+    }
+  }
+
+  override def deleteResource(resourceTypeName: SamResourceTypeName, resourceId: String, userInfo: UserInfo): Future[Boolean] = {
+    val url = samServiceURL + s"/api/resource/${resourceTypeName.value}/$resourceId"
+    val httpRequest = Delete(url)
+    val pipeline = addAuthHeader(userInfo) ~> sendReceive
+    val result: Future[HttpResponse] = pipeline(httpRequest)
+
+    retry(when500) { () =>
+      result.map { response =>
+        response.status match {
+          case s if s.isSuccess => true
+          case _ => false
+        }
+      }
+    }
+  }
+
+  override def userHasAction(resourceTypeName: SamResourceTypeName, resourceId: String, action: SamResourceAction, userInfo: UserInfo): Future[Boolean] = {
+    // special RootJsonReader because DefaultJsonProtocol.BooleanJsonFormat is not root and the implicit
+    // conversion to an Unmarshaller needs a root
+    implicit val rootJsBooleanReader = new RootJsonReader[Boolean] {
+      override def read(json: JsValue): Boolean = DefaultJsonProtocol.BooleanJsonFormat.read(json)
+    }
+
+    val url = samServiceURL + s"/api/resource/${resourceTypeName.value}/$resourceId/action/${action.value}"
+    retry(when500) { () => pipeline[Boolean](userInfo) apply Get(url) }
+  }
+
+  override def overwritePolicy(resourceTypeName: SamResourceTypeName, resourceId: String, policyName: String, policy: SamPolicy, userInfo: UserInfo): Future[Boolean] = {
+    val url = samServiceURL + s"/api/resource/${resourceTypeName.value}/$resourceId/policies/$policyName"
+    val httpRequest = Put(url, policy)
+    val pipeline = addAuthHeader(userInfo) ~> sendReceive
+    val result: Future[HttpResponse] = pipeline(httpRequest)
+
+    retry(when500) { () =>
+      result.map { response =>
+        response.status match {
+          case s if s.isSuccess => true
+          case _ => false
+        }
+      }
+    }
+  }
+
+  override def addUserToPolicy(resourceTypeName: SamResourceTypeName, resourceId: String, policyName: String, memberEmail: String, userInfo: UserInfo): Future[Boolean] = {
+    getResourcePolicies(resourceTypeName, resourceId, userInfo).flatMap { resourcePolicies =>
+      val targetPolicy = resourcePolicies.find(_.policyName.equalsIgnoreCase(policyName)).getOrElse(throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"policy $policyName not found")))
```

Comment:
in the add user case, it should actually create a new policy if the old one wasn't found. we only create the owner policy by default

Reply:
it would probably be better to implement add on the sam side